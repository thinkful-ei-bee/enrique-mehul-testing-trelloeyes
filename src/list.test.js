import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import renderer from 'react-test-renderer';


it('renders without crashing', () => {

    const obj = {header: 'myHeader',
                cards: ['a','b','c']    };

    const div = document.createElement('div');
    ReactDOM.render(
      <List header= {obj.header} cards={obj.cards} />,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
   
  });

  it('get back proper card when returned ', () => {
    const obj = {header: 'myHeader',
                cards: [{id: 'a', title: 'First card', content: 'lorem ipsum'},
                        {id: 'b', title: 'First card', content: 'lorem ipsum'},
                        {id: 'c', title: 'First card', content: 'lorem ipsum'}]   };
    const tree = renderer
      .create(<List header= {obj.header} cards={obj.cards} />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });