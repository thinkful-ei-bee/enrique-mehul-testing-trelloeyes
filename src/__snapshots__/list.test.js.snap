// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get back proper card when returned  1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      myHeader
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
